jbmc/src/java_bytecode/java_object_factory.cpp

```diff
@@ -1435,20 +1435,25 @@ void java_object_factoryt::gen_nondet_array_init(
 ///     int i = nondet(int);
 ///     assume(0 < = i < $VALUES.length);
 ///     expr = $VALUES[i];
-/// where $VALUES is a variable generated by the Java compiler that stores
-/// the array that is returned by Enum.values().
-void java_object_factoryt::gen_nondet_enum_init(
+/// where $VALUES is a variable generated by the Java compiler and which gives
+/// the possible values of a particular enum subtype (this is the same array
+/// that is returned by Enum.values()).
+/// This may fail if the $VALUES static field is not present. Ensuring that
+/// field is in the symbol table when this method may be applicable is the
+/// driver program's responsibility: for example, \ref ci_lazy_methods.cpp makes
+/// some effort to retain this field when a stub method might refer to it.
+/// \return true on success
+bool java_object_factoryt::gen_nondet_enum_init(
   code_blockt &assignments,
   const exprt &expr,
   const java_class_typet &java_class_type,
   const source_locationt &location)
 {
   const irep_idt &class_name = java_class_type.get_name();
   const irep_idt values_name = id2string(class_name) + ".$VALUES";
-  INVARIANT(
```

Comment:
:bulb: An exception might be clearer than a boolean here since it is an error condition, even if we do want to gracefully recover from it.